src/services/F2fService.ts

```diff
@@ -234,6 +234,48 @@ export class F2fService {
 		return sessionItem.Items[0] as ISessionItem;
 	}
 
+	async getSessionsByAuthSessionState(authSessionState: string | undefined): Promise<Array<Record<string, any>>> {
+		const params: QueryCommandInput = {
+			TableName: this.tableName,
+			IndexName: Constants.ATUH_SESSION_STATE_INDEX_NAME,
+			KeyConditionExpression: "authSessionState = :authSessionState",
+			ExpressionAttributeValues: {
+				":authSessionState": authSessionState,
+			},
+		};
+
+		const sessionItems = await this.dynamo.query(params);
+
+		if (!sessionItems?.Items || sessionItems?.Items.length < 1) {
+			this.logger.error("Error retrieving Sessions by authSessionState", {
+				messageCode: MessageCodes.FAILED_FETCHING_SESSION_BY_AUTH_SESSION_STATE,
+			});
+			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error retrieving Sessions by authSessionState");
+		}
```

Comment:
Do we need to do an expired check here to make sure we're not sending emails for expired sessions that haven't yet been garbage collected?

Reply:
Done